### PR TITLE
System Status Report: Add UI for system status report screen

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #5071 

### Description

This PR add new screen for system status report.
The screen uses a view model to fetch system status from remote. Error handling has been added, but the parsing for content will be updated in a separate PR.

### Testing instructions

- Log in to a valid account
- On My Store tab, navigate to Settings screen.
- Select Help & Support > System Status Report.
- Notice that you'll be navigated to System Status Report screen with a loading indicator indicating that the status is being fetched. If an error occurs during this step, an alert will be displayed with option to retry fetching.

### Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/HelpAndSupportViewController.swift [hunk 1]
```diff
@@ -373,7 +373,15 @@ private extension HelpAndSupportViewController {
     /// System status report action
     ///
     func systemStatusReportWasPressed() {
-        // TODO: Show system status view
+        guard let site = ServiceLocator.stores.sessionManager.defaultSite else {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/HelpAndSupportViewController.swift]: nit: how about getting the site ID from `ServiceLocator.stores.sessionManager.defaultStoreID`? when launching a site from login / store switching, the store ID (`defaultStoreID`) is set first then the site (`defaultSite`) is fetched

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/SystemStatusReport/SystemStatusReportViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ///
+    private let stores: StoresManager
+
+    /// Formatted system status report to be displayed on-screen
+    ///
+    @Published private(set) var statusReport: String = ""
+
+    /// Whether fetching system status report failed
+    ///
+    @Published var errorFetchingReport: Bool = false
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/SystemStatusReport/SystemStatusReportViewModel.swift]: nit: can this be `private(set)` as well?

**Review by jaclync — APPROVED** [review]
LGTM :shipit: 

I noticed that the SSR from some sites failed to parse:

- Site 1
```
<> Mapping Error: keyNotFound(CodingKeys(stringValue: "block", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "data", intValue: nil), CodingKeys(stringValue: "pages", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"block\", intValue: nil) (\"block\").", underlyingError: nil))
```
- Site 2 (Jurassic Ninja site)
```
<> Mapping Error: keyNotFound(CodingKeys(stringValue: "subscription", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "data", intValue: nil), CodingKeys(stringValue: "settings", intValue: nil), CodingKeys(stringValue: "taxonomies", intValue: nil)], debugDescription: "No value associated with key CodingKeys(stringValue: \"subscription\", intValue: nil) (\"subscription\").", underlyingError: nil))
```

Maybe it's safer to mark the properties optional? 😅

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/HelpAndSupportViewController.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/HelpAndSupportViewController.swift]: Thanks for catching this! I updated it in 2f52ab2150766c7e6b320e47fe5e49e631a5bd55.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/SystemStatusReport/SystemStatusReportViewModel.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/SystemStatusReport/SystemStatusReportViewModel.swift]: Thank you! This has been fixed in 679146fd9f14092294540fc713a0e4dbef637b5c.

**Comment by itsmeichigo** [comment]
Thank you @jaclync for the review ❤️ As for the parsing issues, I'll update in the next PR to avoid conflicts since I'm already updating the properties' access control. I'll proceed to merge this after CI passes!

**Pull request merged**